.github/workflows/main.yml

```diff
@@ -0,0 +1,29 @@
+name: SpackBot CI
+
+on:
+  pull_request: []
+
+jobs:
+  formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup black environment
+        run: |
+          conda create -c conda-forge -y -q --name black black
```

Comment:
Is conda already installed on GitLab CI? I have my own repo where I'm using an action to set up conda and it takes several minutes to run, so this could be good to know. I wonder if it's also on Windows (where I specifically need it)

Reply:
Anyway, is there a reason to use conda instead of pip? Black doesn't have any non-Python dependencies so we can easily install a wheel with pip.

Reply:
I think so! https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md

Reply:
If I have a choice I think conda is usually better, at least for libraries that have dependencies that might be in other languages. If we extend the CI to do other things it might make sense. But for black alone right now it probably would be the same to use pip. It's about 7 seconds difference in terms of install, with pip being faster on my machine. Here is the conda-forge recipe that shows how it's built https://github.com/conda-forge/black-feedstock/tree/master/recipe there might be subtle differences there too than a vanilla pip install.

Reply:
Looks like they just use pip to install black. I don't think there will be many tools we'll need that aren't pure-Python, but it shouldn't be hard to switch back and forth between pip and conda if we need to.